privx_api/base.py

```diff
@@ -43,9 +43,13 @@ def __exit__(self, exc_type, exc_val, exc_tb) -> None:
         self._connection.close()
 
     def connect(self) -> http.client.HTTPSConnection:
-        return http.client.HTTPSConnection(
-            self.host, port=self.port, context=self.get_context()
-        )
+        try:
+            connection = http.client.HTTPSConnection(
+                self.host, port=self.port, context=self.get_context()
+            )
+        except (OSError, HTTPException) as e:
+            raise InternalAPIException(str(e))
```

Comment:
Usually you add some extra error message as this networking code is deep in the stack: failed to connect to target / failed to create HTTPS client for target etc. Normally when putting stuff in try-except block you want to minimize exception space. So you can also move get_context() out of the try-catch block. to isolate the network exception.

The SSL context, if throwing exception, could be handled separately or could be left as is. In the traceback it would be clear which block causes the exception.